### PR TITLE
Update to gradle 8.7

### PR DESCRIPTION
**Pull request by staktrace**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -173,6 +173,10 @@ subprojects {
   ) {
     apply(plugin = "com.palantir.revapi")
 
+    tasks.named { it == "revapi" || it == "revapiAnalyze" }.configureEach {
+      notCompatibleWithConfigurationCache("revapi is so 2022")
```
Comment (autonomousapps) [inline build.gradle.kts:177]: lol

File: build.gradle.kts [hunk 2]
```diff
@@ -365,10 +373,17 @@ tasks.register("startRedis") {
       "redis-server",
       "--loglevel debug"
     )
-    exec {
-      workingDir(project.rootProject.rootDir.toPath())
+    execOperations.exec {
+      workingDir(rootDir.get().asFile.toPath())
```
Comment (autonomousapps) [inline build.gradle.kts:377]: nit: I think `workingDir()` takes `File` as well, so you could probably drop the `toPath()`

**Review by autonomousapps — COMMENTED** [review]
Looked at one of the failures and I see a lot of issues with Dokka. Are we on latest? I know Dokka has historically had a lot of issues, but I believe they're working on them.
```
See the complete report at file:///home/runner/work/misk/misk/build/reports/configuration-cache/4ukg0qiexrj4ul885cpcdcvnw/4lukq6xhfhwmcti9d72qo3lpy/configuration-cache-report.html
> Invocation of 'Task.project' by task ':detektive:dokkaGfm' at execution time is unsupported.
> Invocation of 'Task.project' by task ':misk-action-scopes:dokkaGfm' at execution time is unsupported.
> Invocation of 'Task.project' by task ':misk-actions:dokkaGfm' at execution time is unsupported.
> Invocation of 'Task.project' by task ':misk-api:dokkaGfm' at execution time is unsupported.
> Invocation of 'Task.project' by task ':misk:dokkaGfm' at execution time is unsupported.
```

**Comment by staktrace** [comment]
I decided for now to make this PR just do the gradle version upgrade and punt on the configuration cache enabling. Given that misk wasn't building with configuration before, and it's a bunch of work to make it configuration-cache-compatible, I'll focus on that separately if we really need it.

**Comment by staktrace** [comment]
https://github.com/cashapp/misk/pull/3266 is a follow-up that enables the configuration cache, but requires deleting the revapi plugin in order to do so

**Review by adrw — APPROVED** [review]

**Pull request merged**